stl/inc/execution

```diff
@@ -1042,6 +1042,10 @@ _Common_diff_t<_InIt1, _InIt2> _Distance_min(_InIt1 _First1, const _InIt1 _Last1
     return _Result;
 }
 
+// This macro does nothing. It serves as a reminder that once _Key is obtained,
+// the amount of work should not be discarded, unless _Key is evaluated as false.
+// Thus any early return that is not due to _Key evaluated as false should be done before obtaining _Key.
+#define _CHUNK_MUST_BE_USED(_Key)

```

Comment:
I like commenting this control flow, but I don't like the do-nothing macro - we have no precedent for this, and macros make life harder for IDEs, clang-format, etc. so we should use them only when necessary.

I think it would be better to replace every occurrence of `_CHUNK_MUST_BE_USED(_Key)` below with the core of your comment here:

```cpp
// Once _Key is obtained, the amount of work should not be discarded.
```

Because this talks about obtaining `_Key` and occurs after testing `if (!_Key)`, I believe that saying `unless _Key is evaluated as false` is unnecessary. The last sentence (beginning with `Thus any early return`) is essentially expanding on the meaning of "discarded". Hmm. Now that the GitHub Issues extension displays hover cards in VSCode (or will, after https://github.com/microsoft/vscode-pull-request-github/issues/1783 is resolved), I additionally suggest:

```cpp
// Once _Key is obtained, the amount of work should not be discarded; see GH-818.
```

I think that will provide sufficient explanation.